ext/iconv/iconv.c

```diff
@@ -758,25 +759,22 @@ static php_iconv_err_t _php_iconv_strlen(size_t *pretval, const char *str, size_
 
 	errno = 0;
 	out_left = 0;
+	more = nbytes > 0;
 
-	for (in_p = str, in_left = nbytes, cnt = 0; in_left > 0; cnt+=2) {
-		size_t prev_in_left;
+	for (in_p = str, in_left = nbytes, cnt = 0; more;) {
 		out_p = buf;
 		out_left = sizeof(buf);
 
-		prev_in_left = in_left;
+		more = in_left > 0;
 
-		if (iconv(cd, (char **)&in_p, &in_left, (char **) &out_p, &out_left) == (size_t)-1) {
-			if (prev_in_left == in_left) {
-				break;
-			}
+		iconv(cd, more ? (char **)&in_p : NULL, more ? &in_left : NULL, (char **) &out_p, &out_left);
```

Comment:
I'm confused about how the error handling here works. Based on reading the man page, I'd expect `in_p` to stay at the same position if there is an illegal multibyte sequence, in which case not handling the error would result in an infinite loop. But that's clearly not what happens...

Reply:
Well, the error handling int ext/iconv is generally confusing, because we're [still supporting `!defined(ICONV_SUPPORTS_ERRNO)`](https://externals.io/message/103125). In the case of invalid multibyte sequences, I don't see an issue though, because not only `in_p` would (eventually) not advance, but also `out_p` would stay (and accordingly the lengths), so the loop would still be terminated (by the `break` statement below).

Reply:
> Well, the error handling int ext/iconv is generally confusing, because we're [still supporting `!defined(ICONV_SUPPORTS_ERRNO)`](https://externals.io/message/103125).

I'll just go ahead and say: Feel free to kill support for it. Just retain a configure error if it's not supported, and assume it is for cross-compilation...

> In the case of invalid multibyte sequences, I don't see an issue though, because not only `in_p` would (eventually) not advance, but also `out_p` would stay (and accordingly the lengths), so the loop would still be terminated (by the `break` statement below).

Thanks, that's what I was missing...